lib/brain.js

```diff
@@ -3037,6 +3051,24 @@ Brain.prototype.returnToCashOutState = function returnToCashOutState () {
   this._screenTimeout(this._chooseFiatCancel.bind(this), 120000)
 }
 
+Brain.prototype.verifyCustomerDiscounts = function verifyCustomerDiscounts () {
+  const discounts = this.trader.individualDiscounts
+
+  if (_.isEmpty(discounts)) return
+  const selectedDiscount = _.find(it => it.customerId === this.customer.id, discounts)
+
+  if (!_.isNil(selectedDiscount)) {
+    /*
+      * It's possible for a customer to insert a promo code before inserting the phone data, leading to two possible discounts.
+      * In that case, use the bigger discount.
+      */
+    const currentDiscount = this.tx.discount || 0
+    if (selectedDiscount.discount > currentDiscount) {
+      this.tx = Tx.update(this.tx, { discount: selectedDiscount.discount, discountSource: 'individualDiscount' })
+    }
```

Comment:
Won't this re-enable the coupon code button in the following scenario?

1. Start the tx with no phone
2. Add a promo code
3. Reache the compliance level of sms verification
4. Have a individual discount bigger than the promo code